packages/eslint-plugin/tests/rules/no-type-alias.test.ts

```diff
@@ -3225,5 +3225,29 @@ type Foo<T> = {
         },
       ],
     },
+    {
+      // unique symbol is not allowed in this context
+      code: 'type Foo = keyof [string] | unique symbol;',
```

Comment:
`unique symbol` can't be used in this context, but we are producing AST tree for this case and validating it in rule.